sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1476,8 +1479,10 @@ MemCpyCommandHost::MemCpyCommandHost(Requirement SrcReq,
     : Command(CommandType::COPY_MEMORY, std::move(DstQueue)),
       MSrcQueue(SrcQueue), MSrcReq(std::move(SrcReq)),
       MSrcAllocaCmd(SrcAllocaCmd), MDstReq(std::move(DstReq)), MDstPtr(DstPtr) {
-  if (!MSrcQueue->is_host())
+  if (!MSrcQueue->is_host()) {
     MEvent->setContextImpl(MSrcQueue->getContextImplPtr());
+    MEvent->setStateIncomplete();
```

Comment:
If I am not misreading this, the `Command` constructor should be setting the state to incomplete when it constructor the event with the `DstQueue`.

Reply:
You are suggesting adding an else clause to this conditional?

If so, I don't think that's necessarily right. An event originally constructed via a queue already has its state set in the event_impl constructor (to either complete or incomplete, depending on whether its a host event, IIRC).  

This just preserves what we had before I made the event context lazy.

Reply:
I'm just saying that this call is redundant as the Command constructor also calls the function for that event.

Reply:
It's not the biggest piece of redundant work, so maybe it's okay. My primary suggestion was to simply just remove the call here as the state would have already been set to Incomplete by the other constructor. In the end, I believe it's much worse that we set the context twice (to different values) which should be more wasted work but less trivial to fix.

Reply:
removed the redundant setStateIncomplete from the subclass